armory/metrics/poisoning.py

```diff
@@ -96,24 +126,31 @@ def get_activations(self, x, batch_size: int = None):
         if batch_size, batch inputs and then concatenate
         """
         activations = []
-        with torch.no_grad():
-            if batch_size:
-                batch_size = int(batch_size)
-                if batch_size < 1:
-                    raise ValueError("batch_size must be false or a positive int")
-            else:
-                batch_size = len(x)
-
-            for i in range(0, len(x), batch_size):
-                x_batch = x[i : i + batch_size]
+        if batch_size:
+            batch_size = int(batch_size)
+            if batch_size < 1:
+                raise ValueError("batch_size must be false or a positive int")
+        else:
+            batch_size = len(x)
+
+        for i in range(0, len(x), batch_size):
+            x_batch = x[i : i + batch_size]
+
+            if self.model_framework == "pytorch":
+                with torch.no_grad():
+                    x_batch = self.preprocess(x_batch)
+                    activation, _ = self.explanatory_model(x_batch)
+                    activations.append(activation.detach().cpu().numpy())
+
+            elif self.model_framework == "tensorflow":
```

Comment:
what's supposed to happen if `model_framework` is neither `pytorch` or `tensorflow`? Should we raise a ValueError?

Reply:
Error raised.